### ISSUE TITLE
iLibCall fails on 64-bt Python versions

### ISSUE THREAD

**Issue by kadler** [IBM/python-itoolkit#17]
**[Original report](https://bitbucket.org/litmis/python-itoolkit/issue/17) by Kevin Adler (Bitbucket: [kadler](https://bitbucket.org/kadler), GitHub: [kadler](https://github.com/kadler)).**

----------------------------------------

All calls to XMLSERVICE through iLibCall fail with `*NODATA`. Other transports are unaffected.

Example:

```
#!python
from itoolkit import *
from itoolkit.lib.ilibcall import *

itransport = iLibCall()
itool = iToolKit()

itool.add(iCmd5250('wrkactjob', 'WRKACTJOB'))
print(itool.xml_in())
itool.call(itransport)
wrkactjob = itool.dict_out('wrkactjob')
print(wrkactjob)
```

Output:
```
#!text
<?xml version='1.0'?>
<xmlservice><sh error="fast" var="wrkactjob"><![CDATA[/QOpenSys/usr/bin/system WRKACTJOB]]></sh></xmlservice>

{'error': {'error': '*NODATA', 'wrkactjob': {...`
```

**Comment by kadler** [comment]
New XMLSERVICE bug is https://github.com/IBM/xmlservice/issues/17

**Comment by dmabupt** [comment]
> **Original comment by Kevin Adler (Bitbucket: [kadler](https://bitbucket.org/kadler), GitHub: [kadler](https://github.com/kadler)).**
> 
> This is an upstream XMLSERVICE bug: https://bitbucket.org/inext/xmlservice-rpg/issues/17/calling-xmlservice-from-a-64-bit-pase-job
> 
> For now, the workaround is to use the database transport:
> 
> ```
> #!python
> # Instead of this
> from itoolkit.lib.ilibcall import *
> itransport = iLibCall()
> 
> # Do this
> from itoolkit.db2.idb2call import *
> itransport = iDb2Call()
> ```
@kadler 
It should be **uppercased B** ->`itransport = iDB2Call()` , not `iDb2Call()`

**Comment by kadler** [comment]
@jkyeung That would make sense to do, if all the `i*Call` transports weren't going to be deprecated in the next release anyway.

**Comment by ramstein74** [comment]
Hello Kevin
this is my code :
from itoolkit import * 
from itoolkit.db2.idb2call import *
itransport = iDB2Call()
itool = iToolKit()

itool.add(iCmd('addlib1', 'addlible SICGA00F55'))
itool.add(iCmd('addlib2', 'addlible $$CLI00F55'))
itool.add(iCmd('addlib3', 'addlible $$CLI00O'))
itool.add(iCmd('callpgm', "CALL PGM($$CLI00O/OC005) PARM(X'001F' X'0056217F' ' ' )"))
itool.call(itransport)
data = itool.dict_out()
print(data)


this is the result :
4.3:zeta@~/pyWork> python3 rpg1.py
{'addlib1': {'success': '+++ success addlible SICGA00F55'}, 'addlib2': {'success': '+++ success addlible $$CLI00F55'}, 'addlib3': {'success': '+++ success addlible $$CLI00O'}, 'callpgm': {'success': "+++ success CALL PGM($$CLI00O/OC005) PARM(X'001F' X'0056217F' ' ' )"}}

all i see is success , however the call
CALL PGM($$CLI00O/OC005) PARM(X'001F' X'0056217F' ' ' )
does not call the programm OC005.
How can i debug it ?

Also i tried this approach because change the call to my program to :

itool.add(
 iPgm('my_key','OC005')
 .addParm(iData('gcnusr','1','2p0'))
 .addParm(iData('gcnuec','56217','6p0'))
 .addParm(iData('$status','1a',' '))
)
always returns error

any help ?

Regards

**Comment by kadler** [comment]
@ramstein74 Seems like a different issue than this one - please open a separate issue.

**Comment by richardschoen** [comment]
Whew, good thing this was documented. I was banging my head trying to get iLibCall to work today :-)

**Comment by kadler** [comment]
This is fixed in XMLSERVICE 2.0.1. The necessary PTFs are now documented in [here](https://python-itoolkit.readthedocs.io/en/latest/api.html#direct-memory-transport) an the code will now attempt to detect this condition and raise an exception with the PTF numbers in the exception message.
